### PR TITLE
Pkg 4697 12.4 initial

### PR DESCRIPTION
**Pull request by danpetry**
**Destination channel:** defaults

### Links

- [PKG-4697](https://anaconda.atlassian.net/browse/PKG-4697)

### Explanation of changes:

- Part of CUDA 12.4 build, following [conda-forge's pattern](https://github.com/conda-forge/staged-recipes/issues/21382) which was developed in conjunction with Nvidia. There isn't a need to stage these then release them all at once. They won't cause a problem released one at a time.
- Please check the main branch too, this has just been forked
- Changes made to adjust feedstocks for differences between conda-forge and defaults. Please see commit messages for description of changes.

[PKG-4697]: https://anaconda.atlassian.net/browse/PKG-4697?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by skupr-anaconda — APPROVED** [review]
It's quite unusual to see a noarch package and compiler stuff at runtime. Is it somewhere described in docs to learn more about it?

File: recipe/meta.yaml [hunk 1]
```diff
@@ -13,13 +13,15 @@ source:
 build:
   number: 0
   noarch: generic
+  skip: true # [not ((linux and x86_64) or win)]
```
Comment (cbouss) [inline recipe/meta.yaml]:
```suggestion
  skip: true # [not (osx or s390x)]
```
Because we still want to test linux-aarch64

**Comment by danpetry** [comment]
@skupr-anaconda in general the reasoning is in the conda-forge recipe in some issue or PR, eg. https://github.com/conda-forge/cuda-compiler-feedstock/pull/3 gives some info
also there's a lot of discussion here https://github.com/conda-forge/conda-forge.github.io/issues/1963

**Comment by danpetry** [comment]
in this case, it's a metapackage, so it's just a way to pull in other packages when used (which need to be run dependencies because otherwise it won't do that). It doesn't need to be arch-specific. It just happens to be one that groups a bunch of compilers.

**Review by cbouss — APPROVED** [review]

**Pull request merged**